### ISSUE TITLE
CI: Many jobs fail cloning repos from opendev.org

### ISSUE THREAD

**Issue by mandre** [gophercloud/gophercloud#2565]
Many CI jobs fail with GnuTLS error, for example:
```
+ functions-common:git_timed:639           :   timeout -s SIGINT 0 git clone https://opendev.org/openstack/glance.git /opt/stack/glance --branch stable/train
Cloning into '/opt/stack/glance'...
error: RPC failed; curl 56 GnuTLS recv error (-110): The TLS connection was non-properly terminated.
fatal: expected 'packfile'
```

https://github.com/gophercloud/gophercloud/actions/runs/4248650726/jobs/7388053337

Many of the answers on stackoverflow for this error message point that it's a known issue with git compiled against gnutls and suggest compiling it against openssl. The github action image installs git from the [git-core ppa](https://launchpad.net/~git-core/+archive/ubuntu/ppa/) and is using gnutls. The official package from the ubuntu repositories also uses gnutls.
We might want to build git ourselves, however that seems odd, instead let's try to use github mirrors to pull the code instead of opendev.org.

**Comment by mandre** [comment]
Switching devstack-action to use github repos by default with https://github.com/EmilienM/devstack-action/pull/12.

**Comment by EmilienM** [comment]
@mandre your PR landed and I released v0.10.0, we can update our CI jobs if you want.
